hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -645,6 +647,55 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 
 	createOrUpdate := r.createOrUpdate(req)
 
+	// Convert old configuration fields to new fields
+
+	if hcluster.Spec.Configuration != nil && len(hcluster.Spec.Configuration.Items) > 0 {
```

Comment:
Shouldn't we also clear the globalconfig and error if both globaconfig and .spec.configuration for a given type are set?

Right now if someone uses the globalconfig we copy it over, if they then change the globalcnifig we won't without any kind of feedback, that is very unintuitive

Reply:
Hmm, can we just prevent any changes to the old config via validation?

Reply:
I will implement as you suggest, the validation is just for additional UX

Reply:
So we are okay with copying and clearing the first time around and then just not propagating any further changes to the global config at all without giving feedback? I would have added some kind of error or condition if both globalconfig and the new field are set for a given config